pkg/model/components/addonmanifests/dnscontroller/remap.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dnscontroller
+
+import (
+	"fmt"
+
+	"github.com/blang/semver/v4"
+	corev1 "k8s.io/api/core/v1"
+	addonsapi "k8s.io/kops/channels/pkg/api"
+	"k8s.io/kops/pkg/featureflag"
+	"k8s.io/kops/pkg/kubemanifest"
+	"k8s.io/kops/pkg/model"
+	"k8s.io/kops/pkg/model/iam"
+)
+
+// Remap remaps the dns-controller addon
+func Remap(context *model.KopsModelContext, addon *addonsapi.AddonSpec, objects []*kubemanifest.Object) error {
+	if !featureflag.UseServiceAccountIAM.Enabled() {
```

Comment:
Consider coding this as `context.UseServiceAccountIAM()`, to ease later transition from feature flag to field in cluster spec.

Reply:
Done - good call!